mesonbuild/compilers/compilers.py

```diff
@@ -53,7 +53,7 @@
 # First suffix is the language's default.
 lang_suffixes = {
     'c': ('c',),
-    'cpp': ('cpp', 'cppm', 'cc', 'cxx', 'c++', 'hh', 'hpp', 'ipp', 'hxx', 'ino', 'ixx', 'C', 'H'),
+    'cpp': ('cpp', 'cppm', 'cc', 'cp', 'cxx', 'c++', 'hh', 'hp', 'hpp', 'ipp', 'hxx', 'h++', 'ino', 'ixx', 'CPP', 'C', 'HPP', 'H'),
```

Comment:
There's some special handling for the capitalized variants somewhere else (we give a warning about case insensitive file systems), so any new capitalized extensions need to be added there as well

Reply:
For some reason I thought there was a warning about case-insensitive file systems in there.